### PR TITLE
Add support for #{input_name} token to regex_find_replace

### PR DESCRIPTION
**Pull request by chambm**
Allows inserting the dataset's display name into the dataset itself

### CODE REVIEW

File: tools/regex_find_replace/test-data/replace3.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+This is a file named find1.txt.
+
```
Comment (bgruening) [inline tools/regex_find_replace/test-data/replace3.txt]: :)

File: tools/regex_find_replace/regex.xml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <tool id="regex1" name="Regex Find And Replace" version="0.1.0">
   <description></description>
-  <command interpreter="python">regex.py --input $input --output $out_file1
+  <command interpreter="python">regex.py --input $input --output $out_file1 --input_display_name $input.display_name
```
Comment (bgruening) [inline tools/regex_find_replace/regex.xml]: quoting of the display name?

Comment (bgruening) [inline tools/regex_find_replace/regex.xml]: and the output file and input

**Comment by chambm** [comment]
@jj-umn Does this look good to you?

**Comment by jj-umn** [comment]
@chambm   
Matt,  let's update the tool versions to: 1.0.0

For a next version, I also was looking at adding an option to parse the input display using named groups and being able to sub those into the replacement strings: 
```
>>> input_display_name = 'Tumor sample.fasta'
>>> input_pattern='^(?P<sample_name>\S+)'
>>> m = re.match(input_pattern,input_display_name)
>>> group_dict = m.groupdict() if m else None
>>> pattern = '>'
>>> replacement = '>generic|\g<sample_name>_'
>>> for key, value in group_dict.items():
...   pat = '\g<%s>' % key
...   replacement = replacement.replace(pat,value)
... 
>>> print replacement
>generic|Tumor_
>>> line = '>NP_997617|0|NM_212452|Rxfp1|relaxin receptor 1 precursor'
>>> re.sub(pattern, replacement, line)
'>generic|Tumor_NP_997617|0|NM_212452|Rxfp1|relaxin receptor 1 precursor'

```

**Comment by chambm** [comment]
What would the user interface look like for that?

**Comment by jj-umn** [comment]
@chambm   How can changing the version number cause testing to fail?    I see why people hate to touch anything that momentarily works.

**Comment by chambm** [comment]
Overwrote the quotation mark. :1st_place_medal:

**Comment by jj-umn** [comment]
@chambm  My aging eyes couldn't see that.

regex.xml
  <command><![CDATA[
    python '$__tool_directory__/regex.py' --input $input --output $out_file1
    #if $name_pattern:
      --input_display_name='$input.display_name'
      --input_pattern='$name_pattern'
    #end if
    #for $check in $checks:
    --pattern='$check.pattern' --replacement='$check.replacement'
    #end for
  ]]></command>
  <inputs>
    <param format="txt" name="input" type="data" label="Select lines from"/>
     <param name="name_pattern" type="text" value="" optional="true" label="Get replacement tokens from dataset name">
                <help>
                </help>
                <sanitizer>
                  <valid>
                    <add preset="string.printable"/>
                    <remove value="&#92;" />
                    <remove value="&apos;" />
                  </valid>
                  <mapping initial="none">
                    <add source="&#92;" target="__backslash__" />
                    <add source="&apos;" target="__sq__"/>
                  </mapping>
                </sanitizer>
      </param>

**Comment by bgruening** [comment]
Good that we have testing now :)
Thanks guys!

**Pull request merged**